src/replication.c

```diff
@@ -3852,6 +3852,11 @@ const char *getFailoverStateString() {
     }
 }
 
+/* Does this server have replication enabled (either as a master or a replica)? */
+int hasReplication() {
+    return server.repl_backlog != NULL;
+}
```

Comment:
not sure i'm a fan of this new function
so far lots of places just check the backlog existence as if it's a common knowledge what it means (some probably have comments).
also, seeing this function name i could imagine that it checks `masterhost` and not `repl_backlog`.
and p.s. the backlog remains alive and maintained even when all replicas dropped (for an hour after they drop).
if we keep it, maybe it should be called `partOfRplicationChain` or negate to `isIsolatedMaster`.

but anyway, i think what you really care for in your case is if the backlog is maintained or not (you don't really care about replicas or masters), just to know if repl_offset is maintained.
so i think you can just look at `repl_backlog` and if you insist on creating a function it can be named `isTrackingReplOffset`

Reply:
I've no problem with skipping the function and just checking `repl_backlog` directly. I'll make the change.